consensus/src/consensusdb/mod.rs

```diff
@@ -122,51 +131,56 @@ impl ConsensusDB {
                 batch.delete::<QCSchema>(hash)
             })
             .collect::<Result<_>>()?;
-        self.commit(batch)
+        self.commit(batch)?;
+        Ok(())
     }
 
     /// Write the whole schema batch including all data necessary to mutate the ledger
     /// state of some transaction by leveraging rocksdb atomicity support.
-    fn commit(&self, batch: SchemaBatch) -> Result<()> {
-        self.db.write_schemas(batch)
+    fn commit(&self, batch: SchemaBatch) -> Result<(), DbError> {
+        self.db.write_schemas(batch)?;
```

Comment:
so commit returns a DbError which means your code above is ... unnecessary... mind you, you can do that cute `map_err` ehre if the write_schemas is returning some other error.